build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchTestBasePlugin.java

```diff
@@ -127,6 +127,9 @@ public void execute(Task t) {
             );
 
             test.getJvmArgumentProviders().add(new SimpleCommandLineArgumentProvider("-XX:HeapDumpPath=" + heapdumpDir));
+            if (test.getJavaVersion().compareTo(JavaVersion.VERSION_24) < 0) {
```

Comment:
@elastic/es-delivery this change does what I want, allowing tests to start under JDK 24, but breaks other tasks in a suprising way (to me):
```
Could not determine the dependencies of task ':libs:entitlement:check'.
> Could not create task ':libs:entitlement:test23'.
   > The value for task ':libs:entitlement:test23' property 'javaLauncher' is final and cannot be changed any further.
```
Seems like mrjar + conditional buinding of the jvmArgs somehow break? I have no clue though 😄

Reply:
I don't think we should use `test.getJavaVersion` here. We should use `buildParams.runtimeJavaVersion`.

Reply:
That wouldn't help with MRJARs right? Should this be more dynamic, looking at the test java version inside a command line argument provider?

Reply:
Good point. Yes, you're right. I was thinking there might be a configuration ordering concern here but probably not.

Reply:
> I was thinking there might be a configuration ordering concern here but probably not.

There is: trying to access the JavaVersion from `test` leads to a premature materialization of the JavaLauncher: 
```
public JavaVersion getJavaVersion() {
   return JavaVersion.toVersion(getJavaLauncher().get().getMetadata().getLanguageVersion().asInt());
}
```

I change to buildParams, and made this dynamic by moving the test inside the command line argument provider as Ryan suggested